### PR TITLE
docs: fix various misleading comments

### PR DESCRIPTION
**Pull request by glozow**
Came across a few misleading comments, wanted to fix them

### CODE REVIEW

File: src/chainparams.h [hunk 1]
```diff
@@ -64,8 +64,9 @@ struct ChainTxData {
 
 /**
  * CChainParams defines various tweakable parameters of a given instance of the
- * Bitcoin system. There are three: the main network on which people trade goods
- * and services, the public test network which gets reset from time to time and
+ * Bitcoin system. There are four: the main network on which people trade goods
+ * and services, the public test network which gets reset from time to time,
+ * a signet test network with extra block requirements (see BIP325) and
  * a regression test mode which is intended for private networks only. It has
  * minimal difficulty to ensure that blocks can be found instantly.
```
Comment (MarcoFalke) [inline src/chainparams.h:71]: nit: It might be better to move the explanation of each network to where each network is defined

Comment (jnewbery) [reply src/chainparams.h:71]: Agree. Better not to enumerate things here which can be documented where they're defined.

File: src/validation.cpp [hunk 2]
```diff
@@ -1099,7 +1100,7 @@ static MempoolAcceptResult AcceptToMemoryPoolWithTime(const CChainParams& chainp
     assert(std::addressof(::ChainstateActive()) == std::addressof(active_chainstate));
     const MempoolAcceptResult result = MemPoolAccept(pool, active_chainstate).AcceptSingleTransaction(tx, args);
     if (result.m_result_type != MempoolAcceptResult::ResultType::VALID) {
-        // Remove coins that were not present in the coins cache before calling ATMPW;
+        // Remove coins that were not present in the coins cache before MemPoolAccept;
```
Comment (jnewbery) [inline src/validation.cpp:1103]:
```suggestion
        // Remove coins that were not present in the coins cache before calling.AcceptSingleTransaction();
```

?

**Review by jnewbery — COMMENTED** [review]
ACK 268965da59bb35083faa4e697bfab3e9b53efbfd

Definitely improvements, but I've added one suggestion. I also agree with Marco about not documenting all of the chains in chainparams.h.

File: src/chainparams.h [hunk 3]
```diff
@@ -64,8 +64,9 @@ struct ChainTxData {
 
 /**
  * CChainParams defines various tweakable parameters of a given instance of the
- * Bitcoin system. There are three: the main network on which people trade goods
- * and services, the public test network which gets reset from time to time and
+ * Bitcoin system. There are four: the main network on which people trade goods
+ * and services, the public test network which gets reset from time to time,
```
Comment (jonatack) [inline src/chainparams.h:68]:
```suggestion
 * and services, the public test network which is reset from time to time,
```

File: src/validation.cpp [hunk 4]
```diff
@@ -690,7 +690,8 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
         }
     }
 
-    // Bring the best block into scope
+    // This is const, but calls into the backend CoinsViews. The CCoinsViewDB at the bottom of the
```
Comment (jonatack) [inline src/validation.cpp:693]:
```suggestion
    // This is const, but calls into the back end CoinsViews. The CCoinsViewDB at the bottom of the
```

**Review by jonatack — COMMENTED** [review]
Concept ACK.

Agree with @MarcoFalke's suggestion. Could squash into one commit. A couple of minor suggestions.

File: src/chainparams.h (hunk 1 above)
Comment (laanwj) [reply src/chainparams.h:71]: Not sure here. I think there can be value in listing short descriptions in one place to have an overview. A more detailed description would belong with the individual chains, sure.

**Comment by glozow** [comment]
Thanks for the reviews, took everyone's suggestions!

File: src/chainparams.h (hunk 1 above)
Comment (glozow) [reply src/chainparams.h]: Ah oops I didn't see this comment! I suppose the issue with listing them here is that people might forget to update it, which is exactly what happened

**Comment by jnewbery** [comment]
ACK 4eca20d6f7

**Comment by MarcoFalke** [comment]
ACK 4eca20d6f7d850492d331d89d1cdd77abb3c70c1

**Comment by laanwj** [comment]
Code review ACK 4eca20d6f7d850492d331d89d1cdd77abb3c70c1

**Pull request merged**